sdk/ai/azure-ai-agents/samples/agents_streaming/sample_agents_stream_eventhandler_with_deep_research_tool.py

```diff
@@ -0,0 +1,151 @@
+# pylint: disable=line-too-long,useless-suppression
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+
+"""
+DESCRIPTION:
+    This sample demonstrates how to use Agent operations with the Deep Research tool from
+    the Azure Agents service, while using an event handler and a synchronous client.
+
+    See also other Deep Research tool samples:
+    - /agents_tools/sample_agents_deep_research.py
+    - /agents_streaming/sample_agents_stream_iteration_with_deep_research_tool.py
+
+    For more information see the Deep Research Tool document: https://aka.ms/agents-deep-research
+
+USAGE:
+    python sample_agents_stream_eventhandler_with_deep_research_tool.py
+
+    Before running the sample:
+
+    pip install azure-ai-projects azure-ai-agents azure-identity
+
+    Set this environment variables with your own values:
+    1) PROJECT_ENDPOINT - The Azure AI Project endpoint, as found in the Overview
+                          page of your Azure AI Foundry portal.
+    2) MODEL_DEPLOYMENT_NAME - The deployment name of the arbitration AI model, as found under the "Name" column in
+       the "Models + endpoints" tab in your Azure AI Foundry project.
+    3) DEEP_RESEARCH_MODEL_DEPLOYMENT_NAME - The deployment name of the Deep Research AI model, as found under the "Name" column in
+       the "Models + endpoints" tab in your Azure AI Foundry project.
+    4) AZURE_BING_CONNECTION_ID - The ID of the Bing connection, in the format of:
+       /subscriptions/{subscription-id}/resourceGroups/{resource-group-name}/providers/Microsoft.MachineLearningServices/workspaces/{workspace-name}/connections/{connection-name}
+"""
+
+import os
+from typing import Any
+from azure.ai.projects import AIProjectClient
+from azure.identity import DefaultAzureCredential
+from azure.ai.agents.models import (
+    AgentEventHandler,
+    DeepResearchTool,
+    MessageRole,
+    MessageDeltaChunk,
+    MessageDeltaTextContent,
+    MessageDeltaTextUrlCitationAnnotation,
+    RunStepDeltaChunk,
+    ThreadMessage,
+    ThreadRun,
+    RunStep,
+)
+
+# When using FunctionTool with ToolSet in agent creation, the tool call events are handled inside the create_stream
```

Comment:
This "FunctionTool with ToolSet in agent creation, the tool call events are handled inside the create_stream" is not fully true anymore, maybe update this or remove